src/basic/log.h

```diff
@@ -367,3 +368,37 @@ int log_syntax_invalid_utf8_internal(
 #define DEBUG_LOGGING _unlikely_(log_get_max_level() >= LOG_DEBUG)
 
 void log_setup(void);
+
```

Comment:
Maybe add a brief comment here, explaining what's this useful for, and how to use it (maybe explicitly mention that the caller owns any memory buffer that gets pushed, and popping doesn't free them)?

Reply:
Added a blurb